server/sonar-server/src/test/java/org/sonar/server/updatecenter/ws/UploadActionTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.updatecenter.ws;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.sonar.core.permission.GlobalPermissions.PROVISIONING;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+
+import com.google.common.io.Resources;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.sonar.server.exceptions.ForbiddenException;
+import org.sonar.server.platform.DefaultServerFileSystem;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.WsActionTester;
+
+public class UploadActionTest {
+
+  @Rule
+  public TemporaryFolder folder = new TemporaryFolder();
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+
+  DefaultServerFileSystem fileSystem = mock(DefaultServerFileSystem.class);
+  File downloadedPluginDir;
+  File plugin = new File(Resources.getResource(getClass(), "UploadActionTest/plugin.jar").getFile());
+
+  @Before
+  public void setUp() throws Exception {
+    downloadedPluginDir = folder.newFolder();
+    when(fileSystem.getDownloadedPluginsDir()).thenReturn(downloadedPluginDir);
+  }
+
+  @Test
+  public void upload_plugin() throws Exception {
+    userSession.login().setGlobalPermissions(SYSTEM_ADMIN);
```

Comment:
suggestion: when all your tests have a specific permission, except to test the check on permission, I'd put the permission setting in the `setUp()` method